pkg/agent/openflow/pipeline_windows.go

```diff
@@ -140,3 +140,96 @@ func (c *client) l3FwdFlowToRemoteViaRouting(localGatewayMAC net.HardwareAddr, r
 }
+
+func (c *client) connectionTrackFlowsAntreaProxyClusterIPSvcHost(category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+
+	connectionTrackCommitTable := c.pipeline[conntrackCommitTable]
+	if c.enableProxy {
+		// These 2 tables are for traffic SNAT of ClusterIP Service from Windows host.
+		serviceConnectionTrackTable := c.pipeline[serviceConntrackTable]
+		serviceConnectionTrackCommitTable := c.pipeline[serviceConntrackCommitTable]
+
+		flows = append(flows,
+			c.pipeline[spoofGuardTable].BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				MatchInPort(config.HostGatewayOFPort).MatchDstIP(config.ServiceGWHairpinIPv4).
+				MatchDstMAC(globalVirtualMAC).
+				Action().GotoTable(serviceConntrackTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			serviceConnectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				Action().CT(false, serviceConnectionTrackTable.GetNext(), SNATCtZone).NAT().CTDone().
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			connectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchCTMark(ServiceCTMark).
+				MatchRegMark(FromGatewayRegMark).
+				Action().GotoTable(serviceConntrackCommitTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			connectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				MatchCTMark(ServiceCTMark).MatchCTStateNew(false).MatchCTStateTrk(true).
+				Action().GotoTable(hairpinSNATTable).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				Done(),
+			serviceConnectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchInPort(config.HostGatewayOFPort).
+				MatchCTStateNew(true).
+				MatchCTStateTrk(true).
+				MatchCTStateDNAT(true).
+				MatchDstMAC(virtualSvcMAC).
+				Action().LoadRegMark(HairpinRegMark).
+				Action().CT(true, serviceConnectionTrackCommitTable.GetNext(), SNATCtZone).
+				SNAT(&binding.IPRange{StartIP: config.ServiceGWHairpinIPv4, EndIP: config.ServiceGWHairpinIPv4}, nil).
+				CTDone().
+				Done(),
+			serviceConnectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+				Cookie(c.cookieAllocator.Request(category).Raw()).
+				MatchInPort(config.HostGatewayOFPort).
+				MatchCTStateNew(false).
+				MatchCTStateTrk(true).
+				Action().CT(false, serviceConnectionTrackCommitTable.GetNext(), SNATCtZone).NAT().
+				CTDone().
+				Done(),
+		)
+	}
+	return flows
+}
+
+func (c *client) l3FwdFlowClusterIPServices(localGatewayMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+	l3FwdTable := c.pipeline[l3ForwardingTable]
+	flows = append(flows, []binding.Flow{
+		l3FwdTable.BuildFlow(priorityLow).MatchProtocol(binding.ProtocolIP).
```

Comment:
Is it possible to be overlap with the default flow to change the dstMAC with antrea-gw0's?

Reply:
Do you mean: `table=70, priority=200,ip,reg0=0x80000/0x80000,nw_dst=192.168.250.1 actions=mod_dl_dst:00:15:5d:1a:d4:02,resubmit(,80)`?

They have different priority and the above flow has `nw_dst=192.168.250.1` match condition which I don't think will match ClusterIP packets?